### PR TITLE
Changes to storage RFC based on previous comments

### PR DESCRIPTION
**Pull request by jtweed**

### CODE REVIEW

File: docs/rfcs/002-archival_storage/README.md [hunk 1]
```diff
@@ -290,15 +290,16 @@ Content-Type: application/json
...
 
 - Check that the supplied version matches the current version
-- Validate and unpack the supplied bag
+- Unpack the supplied bag
 - Store the supplied bag as a new version
-- Update the current version of the bag
+- Register the new version of the bag as the current version
 
-Partial updates, where files that are not changed are not resupplied, are supported through the use of `fetch.txt` in the supplied bag. File references must include the bag version in which the file was previously supplied.
+Partial updates, where files that are not changed are not resupplied, are supported through the use of `fetch.txt` in the supplied bag. File references must specify the full storage location of the previously supplied file, including the version number of the bag in which it was last supplied in the path.
```
Comment (mklander) [inline docs/rfcs/002-archival_storage/README.md]: Perhaps an example `fetch.txt`
> https://s3-eu-west-1.amazonaws.com/wellcomecollection-storage-staging-access/alex-testing/b12345/v3/data/b24923333.xml 98765 data/b24923333.xml

File: docs/rfcs/002-archival_storage/README.md [hunk 2]
```diff
@@ -290,15 +290,16 @@ Content-Type: application/json
...
 
 Updates with fetch files should be processed as follows:
 
-- Check that files in `fetch.txt` exist
+- Check that files in `fetch.txt` reference files in the correct bag
+- Check that files in `fetch.txt` exist most recently at the specified version
```
Comment (mklander) [inline docs/rfcs/002-archival_storage/README.md]: We should make this explicit too, does the fetch as described here only support references to files already in the storage service (which seems a sensible first step)?  If so how should those entries be recognised, just by matching bucket/space/bag. If entries do not match a previous version or attempt to skip a version by pointing to an older version what should be done?  If we are match only against the previous bag version that could be explicit here too -- either match a fetch entry in the previous version or an existing file.

**Comment by tomcrane** [comment]
I have updated the client view of the API described in https://github.com/wellcometrust/platform/issues/3503. The new version is cleaner, and works from my perspective.

**Comment by tomcrane** [comment]
The DDS, and any DDS successor, are version-unaware (or rather, version-don't-care) consumers of the API. They only want the latest, access, view of the assets for a given identifier. 

Clients that do care can use the `versions` listed in the bags response and then make requests for versioned bags by appending `[?version={version}]` (for example, to generate a report on the asset history for an identifier, showing diffs or whatever). So that's OK.

Although the current usage for testing and development of the storage service, and migration, is not a typical production workflow usage of the API, they are real use cases for, well, testing and development. Should the API have support for pre-production scenarios where you want to delete or overwrite bags without version increments?

**Comment by tomcrane** [comment]
A thought

```

      {
        "type": "File",
        "name": "data/objects/b24923333_001.jp2",
        "path": "v1/data/objects/b24923333_001.jp2",
        "checksum": "e68c93a5170837420f63420bd626650b2e665434e520c4a619bf8f630bf56a7e"
      },
```

While the dropping of `bagVersion`, and the inclusion of both `name` and `path` (with the versioning incorporated into the path) means that the _version-don't-care_ client doesn't have to compute paths, would it still be useful for the `File` to have an explicit version property as well, for clients that do need to be aware of versions? e.g., Goobi?

File: docs/rfcs/002-archival_storage/README.md (hunk 1 above)
Comment (jtweed) [reply docs/rfcs/002-archival_storage/README.md]: There's one in the partial bag example later on in the RFC.

**Comment by jtweed** [comment]
> Should the API have support for pre-production scenarios where you want to delete or overwrite bags without version increments?

I'd say no, unless it proves to be really necessary. We intend to fully clear things down as part of the migration, so post-migration everything should start as v1.

**Comment by jtweed** [comment]
> A thought
> 
> ```
> 
>       {
>         "type": "File",
>         "name": "data/objects/b24923333_001.jp2",
>         "path": "v1/data/objects/b24923333_001.jp2",
>         "checksum": "e68c93a5170837420f63420bd626650b2e665434e520c4a619bf8f630bf56a7e"
>       },
> ```
> would it still be useful for the `File` to have an explicit version property as well, for clients that do need to be aware of versions? e.g., Goobi?

I don't want to make it look like we have file level versioning. I am interested in Intranda's feedback on this before we confirm it though. I'm going to merge now so I can send it to them.

File: docs/rfcs/002-archival_storage/README.md (hunk 2 above)
Comment (jtweed) [reply docs/rfcs/002-archival_storage/README.md]: Is that not covered by "exist most recently at the specified version"? If these checks fail, the ingest should fail.

**Comment by tomcrane** [comment]
The scenario would be a client that sweeps through the file list, picking out all those that match a particular bag version (perhaps the current version) and doing something with them. Would Goobi, or something else, ever need to do that kind of thing?

**Pull request merged**